plans/centos8/integration.fmf

```diff
@@ -0,0 +1,7 @@
+discover:
+    how: fmf
+#    filter: 'tag: integration & tag: centos8'
```

Comment:
@psss @thrix could you guys help me to realize what I am doing wrong with this filter. My expected behavior is that the integration plan will pick tests only with these tags

Reply:
Sure, checking what's wrong:

```
> tmt
Found 2 tests: /tests/integration/inhibit-if-kmods-is-not-supported and /tests/unit.
Found 2 plans: /plans/centos8/integration and /plans/centos8/unit.
Found 0 stories.
> tmt test ls
/tests/integration/inhibit-if-kmods-is-not-supported
/tests/unit
> tmt test ls --filter tag:integration

> tmt test show
/tests/integration/inhibit-if-kmods-is-not-supported
     summary kmods tests
        test pytest
        path /tests/integration/inhibit-if-kmods-is-not-supported
      manual no
    duration 5m
     enabled yes
      result respect

/tests/unit
     summary kmods tests
        test pytest .
        path /tests/unit
      manual no
    duration 5m
     enabled yes
      result respect
```

So the reason is that none of your tests has the `tag` defined.
Let's see why:

```
> tmt test show -v integration
/tests/integration/inhibit-if-kmods-is-not-supported
     summary kmods tests
        test pytest
        path /tests/integration/inhibit-if-kmods-is-not-supported
      manual no
    duration 5m
     enabled yes
      result respect
     sources /home/psss/git/github/convert2rhel/tests/integration/main.fmf
             /home/psss/git/github/convert2rhel/tests/integration/inhibit-if-kmods-is-not-supported/main.fmf
      fmf-id name: /tests/integration/inhibit-if-kmods-is-not-supported
             url: https://github.com/oamg/convert2rhel.git
             ref: pr195

> cat /home/psss/git/github/convert2rhel/tests/integration/main.fmf
summary: Integration tests
tags: [integration, centos8]
prepare:
  how: shell
  # TODO install rpm script
  script: |
    echo "Installing rpm here"
```

Ah, I see. The correct name for the attribute is [tag](https://tmt.readthedocs.io/en/latest/spec/tests.html#tag). All test attributes are kept consistent in singular.

Reply:
Also note, that if a particular test needs a package installed on the guest, you can use the [require](https://tmt.readthedocs.io/en/latest/spec/tests.html#require) attribute for this. The [prepare](https://tmt.readthedocs.io/en/latest/spec/steps.html#prepare) step can be used in plans only.

Reply:
thx @psss, do you think it is possible to create a json schema for TMT configs? if yes, than it would be possible to upload it https://github.com/schemastore/schemastore/ and a lot of editors will start to validate fmf  files. Added https://github.com/psss/tmt/issues/602